DXMainClient/DXGUI/GameClass.cs

```diff
@@ -143,10 +152,47 @@ protected override void Initialize()
             ProgramConstants.PLAYERNAME = playerName;
             UserINISettings.Instance.PlayerName.Value = playerName;
 
-            LoadingScreen ls = new LoadingScreen(wm);
-            wm.AddAndInitializeControl(ls);
-            ls.ClientRectangle = new Rectangle((wm.RenderResolutionX - ls.Width) / 2,
-                (wm.RenderResolutionY - ls.Height) / 2, ls.Width, ls.Height);
+            var loadingScreen = BuildContainer(wm).GetInstance<LoadingScreen>();
+            wm.AddAndInitializeControl(loadingScreen);
+            loadingScreen.ClientRectangle = new Rectangle((wm.RenderResolutionX - loadingScreen.Width) / 2,
+                (wm.RenderResolutionY - loadingScreen.Height) / 2, loadingScreen.Width, loadingScreen.Height);
+            
+        }
+
+        private Container BuildContainer(WindowManager windowManager)
+        {
+            var container = new Container();
+
+            container.RegisterSingleton<ServiceProvider>();
+            container.RegisterInstance(windowManager);
+            container.RegisterInstance(GraphicsDevice);
+            container.RegisterSingleton<LoadingScreen>();
+            container.RegisterSingleton<GameCollection>();
+            container.RegisterSingleton<CnCNetUserData>();
+            container.RegisterSingleton<CnCNetManager>();
+            container.RegisterSingleton<TunnelHandler>();
+            container.RegisterSingleton<TopBar>();
+            container.RegisterSingleton<OptionsWindow>();
+            container.RegisterSingleton<PrivateMessagingWindow>();
+            container.RegisterSingleton<PrivateMessagingPanel>();
+            container.RegisterSingleton<LANLobby>();
+            container.RegisterSingleton<CnCNetGameLobby>();
+            container.RegisterSingleton<CnCNetGameLoadingLobby>();
+            container.RegisterSingleton<CnCNetLobby>();
+            container.RegisterSingleton<GameInProgressWindow>();
+            container.RegisterSingleton<SkirmishLobby>();
+            container.RegisterSingleton<MainMenu>();
+            container.RegisterSingleton<MapLoader>();
+            container.RegisterSingleton(() => 
```

Comment:
@Metadorius 
The reason for `NullDiscordHandler`, is because you cannot return a `null `value for this dependency using SimpleInjector. So, I created a dummy handler that would prevent usages of it from breaking the application, but nothing would actually happen. 

Regarding your comment about having to restart the client, you are correct. That would be a requirement in this case.

Reply:
Yes, but why empty handler? Can't that just be handled that from the `DiscordHandler` itself without introducing null handler and requiring restart?

Reply:
It could be. It was more for simplicity, I guess. If the dependency injection returns this empty handler, no other codes requires a change. If it returns a valid DiscordHandler, I was finding that nearly all of the public functions in the DiscordHandler required a null check on the internal client. It just felt very cluttered.

Reply:
Also, in the previous code, in `LoadingScreen.cs`, the DiscordHandler was only ever instantiated if Discord was enabled. Otherwise, it was null. Where it was attempted to be used could be easily null checked on the DiscordHandler instance itself. Like I said though, in my dependency injection, a dependency cannot be returned as null. So, this was just another solution.